java-frontend/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java

```diff
@@ -272,17 +281,29 @@ public CFG cfg() {
   @Override
   @Nullable
   public Boolean isOverriding() {
-    if (isStatic() || isPrivate()) {
-      return false;
-    }
-    if (isAnnotatedOverride()) {
-      return true;
+    if (isOverridingCached) {
+      return isOverriding;
     }
-    List<Symbol.MethodSymbol> overriddenSymbols = symbol().overriddenSymbols();
-    if (overriddenSymbols.isEmpty()) {
-      return false;
+    isOverridingCached = true;
+
+    if (isStatic() || isPrivate()) {
+      isOverriding = false;
+    } else if (isAnnotatedOverride()) {
+      isOverriding = true;
+    } else {
+      Symbol.MethodSymbol symbol = symbol();
+      List<Symbol.MethodSymbol> overriddenSymbols = symbol.overriddenSymbols();
+      if (overriddenSymbols.isEmpty()) {
+        if (hasUnknownTypeInHierarchy(symbol)) {
+          isOverriding = null;
+        } else {
+          isOverriding = false;
+        }
```

Comment:
```suggestion
      if (overriddenSymbols.isEmpty() && !hasUnknownTypeInHierarchy(symbol)) {
        isOverriding = false;
```

Reply:
Not sure about this one, the value null has a meaning in this context. Am I missing something?

Reply:
The value is indeed null by default. My issue is with the number of nested branches that can be simplified either by changing the test in the if statement or doing early returns. WDYT?